.github/ISSUE_TEMPLATE/release.md

```diff
@@ -7,97 +7,108 @@ labels: release
 <!--
 Issue title should be in the following format:
 
-    Cut vX.Y.Z Release
--->
-
-<!--
-Please assign the release manager to the issue.
--->
+    Cut vX.Y.Z Release on DATE
 
-### Release Information
+For example:
 
-**Type:**
-<!-- Select One -->
-- [ ] Minor
-- [ ] Patch
+    Cut v1.3.0 on June 29, 2021.
 
-**Version:** <!-- e.g. v1.3.0 -->
-
-**Release Date:** <!-- e.g. Jun 29, 2021 -->
+Please assign the release manager to the issue.
+-->
 
 ### Steps
+<!--
+Please complete the following steps in order. Links should be populated at the
+bottom of this section.
+-->
 
-<!-- Please complete the following steps in order. Links should be populated at the bottom of this section. -->
+This issue can be closed when we have:
+
+<!--
+Please uncomment the top section only if cutting a minor release. These
+should be completed at the beginning of Code Freeze:
+https://crossplane.io/docs/v1.2/reference/release-cycle.html#code-freeze
+-->
 
-I have:
-<!-- Please uncomment the following section only if cutting a minor release. These should be completed at the beginning of Code Freeze: https://crossplane.io/docs/v1.2/reference/release-cycle.html#code-freeze -->
 <!--
 - [ ] Created the [release branch][release-branch].
-- [ ] Created and merged an empty commit to the `master` branch
-  ([PR][empty-commit-pr]).
-- [ ] Run the [Tag workflow][rc-tag-workflow] on the `master` branch with the
-  next release candidate tag ([Tag][rc-tag]).
+- [ ] Created and merged an empty commit to the `master` branch ([PR][empty-commit-pr]).
```

Comment:
```suggestion
- [ ] Created and merged an empty commit to the `master` branch ([link][empty-commit-pr]).
```

I considered making this change too, so that it's super clear to readers which link in each bullet point will take them to the relevant artifact. Another option would be to ask assignee to simply paste links to the artifact of each step as comments. That would make life more difficult for the reader, but it would make life easier for the assignee. Currently you must continually edit, save, edit, save etc in the markdown of the issue description as you proceed through the release. I would have preferred to simply check off boxes in the issue description as I went and paste links to the relevant artifacts as distinct comments.

Reply:
@negz I typically do this by copying the markdown into my editor and making changes there, rather than trying to do it on the issue, which github does not give a great interface for. I would prefer not opting to use comments, but I am fine with changing the name of the link to `[link]`.

Reply:
Hrm, I'd personally prefer to lean into a flow that's more compatible with the native GitHub interface but I don't feel that strongly about it. Maybe we can have a few other folks work through the template and get feedback from them. For now I'll just make the `[link]` change.

Reply:
Are you saying that adding comments is more leaning into the GitHub interface flow? Since we can't have templates in comments I feel like this is a recipe for divergence from convention, potentially making the issues very difficult to parse when looking back on them.

Reply:
I'm saying needing to leave GitHub and fill out the template in my IDE, then copy and paste it back into GitHub when I'm done does not seem like leaning into the GitHub UX very much. I can't imagine it's common or intuitive for folks to work that way with issues.

That said, this is not a big deal. I'm happy to defer until we have more folks with experience and opinions here.